docs/src/_docs/advanced.md

```diff
@@ -143,34 +143,39 @@ using Turing
 
 # Define the simple gdemo model.
 @model gdemo(x, y) = begin
- s ~ InverseGamma(2,3)
- m ~ Normal(0,sqrt(s))
- x ~ Normal(m, sqrt(s))
- y ~ Normal(m, sqrt(s))
- return s, m
+    s ~ InverseGamma(2,3)
+    m ~ Normal(0,sqrt(s))
+    x ~ Normal(m, sqrt(s))
+    y ~ Normal(m, sqrt(s))
+    return s, m
+end
+
+function get_nlogp(model)
+    # Set up the model call, sample from the prior.
+    vi = Turing.VarInfo(model)
+
+    # Define a function to optimize.
+    function nlogp(sm)
+        spl = Turing.SampleFromPrior()
```

Comment:
Maybe add an explanation here why this is more efficient.

Reply:
It's just that it uses the `TypedVarInfo` instead of the `UntypedVarInfo`. I also made sure that no global variables are wrapped in the example, so that will make evaluating the `nlogp` function faster.

Reply:
`Turing.VarInfo()` makes an `UntypedVarInfo`. `Turing.VarInfo(model)` makes a `TypedVarInfo`.

Reply:
I see - thanks. It's worth to add this information to the `advanced` section of our docs. But maybe in a separate PR?

Reply:
Sure